server/src/test/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetServiceTests.java

```diff
@@ -297,12 +320,284 @@ public void testResetStartRequestIfTranslogIsCorrupted() throws Exception {
+                    switch (downloadFileErrorType) {
+                        case CORRUPTED_FILE:
+                            byte[] fileDataCopy = new byte[fileData.length];
+                            System.arraycopy(fileData, 0, fileDataCopy, 0, fileData.length);
+                            // Corrupt the file
+                            for (int i = 0; i < randomIntBetween(1, fileDataCopy.length); i++) {
+                                fileDataCopy[i] = randomByte();
+                            }
+                            return new ByteArrayInputStream(fileDataCopy);
+                        case TRUNCATED_FILE:
+                            final int truncatedFileLength = fileData.length / 2;
+                            byte[] truncatedCopy = new byte[truncatedFileLength];
+                            System.arraycopy(fileData, 0, truncatedCopy, 0, truncatedFileLength);
+                            return new ByteArrayInputStream(truncatedCopy);
+                        case LARGER_THAN_EXPECTED_FILE:
+                            byte[] largerData = new byte[fileData.length + randomIntBetween(1, 250)];
+                            System.arraycopy(fileData, 0, largerData, 0, fileData.length);
+                            for (int i = fileData.length; i < largerData.length; i++) {
+                                largerData[i] = randomByte();
+                            }
+                            return new ByteArrayInputStream(largerData);
+                        case FETCH_ERROR:
+                            throw new RuntimeException("Unexpected error");
+                        default:
+                            throw new IllegalStateException("Unexpected value: " + downloadFileErrorType);
+                    }
+                }
+            };
+
+        recoveryStateIndex.addFileDetail(storeFileMetadata.name(), storeFileMetadata.length(), false);
+        recoveryStateIndex.setFileDetailsComplete();
+
+        RecoveryTarget recoveryTarget = new RecoveryTarget(shard, null, snapshotFilesProvider, null);
+
+        String repositoryName = "repo";
+        IndexId indexId = new IndexId("index", "uuid");
+        BlobStoreIndexShardSnapshot.FileInfo fileInfo =
+            new BlobStoreIndexShardSnapshot.FileInfo("name", storeFileMetadata, SNAPSHOT_FILE_PART_SIZE);
+
+        PlainActionFuture<Void> writeSnapshotFileFuture = PlainActionFuture.newFuture();
+        recoveryTarget.downloadSnapshotFile(repositoryName, indexId, fileInfo, writeSnapshotFileFuture);
+        ExecutionException executionException = expectThrows(ExecutionException.class, writeSnapshotFileFuture::get);
+
+        Throwable downloadFileError = executionException.getCause();
+        switch (downloadFileErrorType) {
+            case CORRUPTED_FILE:
+            case LARGER_THAN_EXPECTED_FILE:
+                // Files larger than expected are catch by VerifyingIndexInput too
```

Comment:
```suggestion
                // Files larger than expected are caught by VerifyingIndexInput too
```

English can be daft sometimes :)

Reply:
🤦